experimental/emulated_vault/opt/emulated-vault/vault-server

```diff
@@ -0,0 +1,249 @@
+            self._do_DELETE()
+        except:
+            logger.exception("do_DELETE exception")
+
+    def _do_GET(self):
+        logger.debug("GET %s", self.path)
+        parsed_path = urlparse.urlparse(self.path)
+
+        if parsed_path.path == "/ping":
+            logger.info("Ping")
+            success, value = adaptor.ping()
+            if not success:
+                self.send_response(503)
+                self.send_header('Content-type', 'text/html')
+                self.end_headers()
+                self.wfile.write("Failure")
+                return
+
+            self.send_response(200)
+            self.send_header('Content-type', 'text/html')
+            self.end_headers()
+            self.wfile.write(value)
+            return
+
+        if parsed_path.path == "/search/query/sslkeys":
+            logger.info("Search SSL: path=%s query=%s", parsed_path.path, parsed_path.query)
+            filters = {}
+            keyFilters = {}
+            cdnFind = re.search(".*q=cdn:([^&]*).*", parsed_path.query)
+            if cdnFind:
+                def cdnFilter(key,val):
+                    try:
+                        data = json.loads(val)
+                        return data['cdn']==cdnFind.group(1)
+                    except:
+                        return False
+                filters['cdn'] = cdnFilter
+
+            dsFind = re.search(".*q=deliveryservice:([^&]*).*", parsed_path.query)
+            if dsFind:
+                def dsFilter(key,val):
+                    try:
+                        data = json.loads(val)
+                        return data['deliveryservice']==dsFind.group(1)
+                    except:
+                        return False
+                filters['ds'] = dsFilter
+
+            keyFind = re.search(".*q=_yz_rk:([^&]*).*", parsed_path.query)
+            if keyFind:
+                def keyFilter(key):
+                    return fnmatch.fnmatch(os.path.basename(key), keyFind.group(1))
+                keyFilters['key-match'] = keyFilter
+
+            success, parameters = adaptor.searchParameters("riak/ssl/", keyFilters=keyFilters, filters=filters)
+            if not success:
+                self.send_response(503)
+                self.send_header('Content-Type', 'application/json')
+                self.end_headers()
+                self.wfile.write('{"Failure"}')
+                return
+
+            docs = []
+            for key,val in parameters.iteritems():
+                docs.append(json.loads(val))
+            toReturn = {"response":{"numFound":len(docs),"start":0, "docs":docs}}
+
+            self.send_response(200)
+            self.send_header('Content-Type', 'application/json')
+            self.end_headers()
+            self.wfile.write(json.dumps(toReturn))
+            return
+
+        logger.info("GET %s", self.path)
+        success, value = adaptor.getParameter(parsed_path.path[1:])
+        if not success:
+            self.send_response(503)
```

Comment:
is this the appropriate response code for not finding a parameter? Should this not be a more generic `500 Internal Server Error` response since we don't know why the value could not be gotten?

Reply:
I wrote it a while ago - need to check what is RIAK behavior. Might have been copied from it

Reply:
Traffic Ops just checks for 200 responses (2XX? not sure), so it shouldn't care. But [the RIAK docs](https://docs.riak.com/riak/kv/2.2.3/developing/api/http/fetch-object.1.html#response)  says that it will return 503 in the event of an internal timeout whereas a key not being found is a `404 Not Found`. This probably will work with ATC today, and it may not ever make a behavioral difference, but even if only for the purposes of logging info it might be better to send the more appropriate `404 Not Found`.